src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -509,31 +552,39 @@ public bool TryUpdate(TKey key, TValue newValue, TValue comparisonValue)
         /// </summary>
         /// <param name="key">The key whose value is compared with <paramref name="comparisonValue"/> and
         /// possibly replaced.</param>
-        /// <param name="hashcode">The hashcode computed for <paramref name="key"/>.</param>
+        /// <param name="nullableHashcode">The hashcode computed for <paramref name="key"/>.</param>
         /// <param name="newValue">The value that replaces the value of the element with <paramref
         /// name="key"/> if the comparison results in equality.</param>
         /// <param name="comparisonValue">The value that is compared to the value of the element with
         /// <paramref name="key"/>.</param>
-        /// <returns>true if the value with <paramref name="key"/> was equal to <paramref
-        /// name="comparisonValue"/> and replaced with <paramref name="newValue"/>; otherwise,
-        /// false.</returns>
-        /// <exception cref="System.ArgumentNullException"><paramref name="key"/> is a null
-        /// reference.</exception>
-        private bool TryUpdateInternal(TKey key, int hashcode, TValue newValue, TValue comparisonValue)
+        /// <returns>
+        /// true if the value with <paramref name="key"/> was equal to <paramref name="comparisonValue"/> and
+        /// replaced with <paramref name="newValue"/>; otherwise, false.
+        /// </returns>
+        /// <exception cref="ArgumentNullException"><paramref name="key"/> is a null reference.</exception>
+        private bool TryUpdateInternal(TKey key, uint? nullableHashcode, TValue newValue, TValue comparisonValue)
         {
-            Debug.Assert(_comparer.GetHashCode(key) == hashcode);
+            IEqualityComparer<TKey>? comparer = _comparer;
+
+            Debug.Assert(
+                nullableHashcode is null ||
+                (comparer is null && key.GetHashCode() == nullableHashcode) ||
+                (comparer != null && comparer.GetHashCode(key) == nullableHashcode));
+
+            uint hashcode =
+                nullableHashcode.HasValue ? nullableHashcode.GetValueOrDefault() :
+                comparer is null ? (uint)key.GetHashCode() :
+                (uint)comparer.GetHashCode(key);
```

Comment:
Is the second line equivalent to `nullableHashcode ??`?

Reply:
Yes, I can simplify it.